### PR TITLE
Added main framework demo

### PR DESCRIPTION
**Pull request by DavidHaighton**

### CODE REVIEW

File: ScreeningStep.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        :param image the image from the camera:
+        :return the status of the step:
+        """
+        runState = 0  # running status
+        if self.__status == ScreeningStep.START:
+            self.start = timeit.default_timer()
+            pass
+        runState = self._run_step(image)
+
+        self.__status = return_val = ScreeningStep.FAIL if self._is_timeout() else runState
```
Comment (Justgo13) [inline ScreeningStep.py:44]: how come there is a return_val in between the two equals?

**Review by NosBos — APPROVED** [review]

File: ScreeningStep.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    def _run_step(self, image) -> int:
+        print(self.output)
+        color = (255, 0, 0)
+        end_point = (220, 220)
+        start_point = (5, 5)
+        thickness = 2
+        image = cv2.rectangle(image, start_point, end_point, color, thickness)
+        return ExampleStep.RUNNING
+
+    pass
```
Comment (Justgo13) [inline ScreeningStep.py:87]: leave this as a comment if it is not being used

File: ScreeningStep.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+class VaccinePassportStep(ScreeningStep):
+
+    def __init__(self):
+        super.__init__("Vaccine Passport", "Scan the QR Code for your Proof of Vaccination", 15)
+        pass
+
+    def _run_step(self, image) -> int:
+        return ScreeningStep.RUNNING
+    pass
```
Comment (Justgo13) [inline ScreeningStep.py]: I feel that each of these classes would be in their own modules?

File: main.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        status = steps[current].run(frame)
+        if status == ScreeningStep.ScreeningStep.FAIL:
+            current = 0  # reset
+        elif status == ScreeningStep.ScreeningStep.SUCCESS and current == len(steps)-1:
+            pass # open door step??
+        elif status == ScreeningStep.ScreeningStep.SUCCESS:
+            current = (current+1) % len(steps)  # proceed to next step
+            pass
+
+        cv2.imshow('frame', cv2.flip(frame, 1))  # show flipped screen for easier time
```
Comment (NosBos) [inline main.py]: Flipping will cause text to turn the incorrect way

**Review by Justgo13 — APPROVED** [review]

**Pull request merged**